Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+
+    private static final String TAG = BaiduMapFragment.class.getSimpleName();
+
+    private static final IntentFilter mEventFilter = new IntentFilter();
+    private LocationClient mBDLocClient;
+    public MyBDLocationListenner mBDLocListener = new MyBDLocationListenner();
+    private final HashBiMap<MarkerInfo, Marker> mBiMarkersMap = new HashBiMap<MarkerInfo, Marker>();
+    private DroidPlannerPrefs mAppPrefs;
+    private final AtomicReference<AutoPanMode> mPanMode = new AtomicReference<AutoPanMode>(AutoPanMode.DISABLED);
+
+    private Polyline mFlightPath;
+    private Polyline mMissionPath;
+    private Polyline mDroneLeashPath;
+    private int maxFlightPathSize;
+
+    private DPMap.OnMapClickListener mMapClickListener;
+    private DPMap.OnMapLongClickListener mMapLongClickListener;
+    private DPMap.OnMarkerClickListener mMarkerClickListener;
+    private DPMap.OnMarkerDragListener mMarkerDragListener;
+    private android.location.LocationListener mLocationListener;
+
+    protected boolean useMarkerClickAsMapClick = false;
+
+    private List<Polygon> polygonsPaths = new ArrayList<Polygon>();
+
+    protected DroidPlannerApp mDpApp;
+    private Polygon mFootprintPoly;
+
+    static {
+        mEventFilter.addAction(AttributeEvent.GPS_POSITION);
+        mEventFilter.addAction(SettingsFragment.ACTION_MAP_ROTATION_PREFERENCE_UPDATED);
+        mEventFilter.addAction(com.baidu.mapapi.SDKInitializer.SDK_BROADTCAST_ACTION_STRING_PERMISSION_CHECK_ERROR);
+        mEventFilter.addAction(com.baidu.mapapi.SDKInitializer.SDK_BROADCAST_ACTION_STRING_NETWORK_ERROR);
+    }
+
+    private final BroadcastReceiver eventReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            final String action = intent.getAction();
+            switch (action) {
+                case AttributeEvent.GPS_POSITION:
+                    if (mPanMode.get() == AutoPanMode.DRONE) {
+                        final Drone drone = getDroneApi();
+                        if (!drone.isConnected())
+                            return;
+
+                        final Gps droneGps = drone.getAttribute(AttributeType.GPS);
+                        if (droneGps != null && droneGps.isValid()) {
+                            final LatLong droneLocation = droneGps.getPosition();
+                            updateCamera(droneLocation);
+                        }
+                    }
+                    break;
+                case SettingsFragment.ACTION_MAP_ROTATION_PREFERENCE_UPDATED:
+                    setupMapUI(getBaiduMap());
+                    break;
+                case com.baidu.mapapi.SDKInitializer.SDK_BROADTCAST_ACTION_STRING_PERMISSION_CHECK_ERROR:
+                    Toast.makeText(getActivity().getApplicationContext(),
+                            "BaiduMap: Keycode error, please check the keycode in AndroidManifest.xml!", Toast.LENGTH_SHORT).show();
```

Comment:
@droneboost use `Toast.LENGTH_LONG` for this toast to give time to the user to read it.